clients/tests/unit/render/util.test.js

```diff
@@ -1,13 +1,226 @@
+
+
+
+
+	test( 'removeFromUploadStarted removes a valid entry when its the only fieldId', () => {
+		cf2.uploadStarted.push( 'fld123');
+		removeFromUploadStarted('fld123', cf2 );
+		expect(cf2.uploadStarted.length).toBe(0);
+	});
+
+	test( 'removeFromUploadStarted removes a valid entry when its one of many fieldId', () => {
+		cf2.uploadStarted.push( 'fld456');
+		cf2.uploadStarted.push( 'fld123');
+		removeFromUploadStarted('fld123', cf2 );
+		expect(cf2.uploadStarted.length).toBe(1);
+		expect( cf2.uploadStarted.includes('fld456')).toBe(true);
+		expect( cf2.uploadStarted.includes('fld123')).toBe(false);
+
+	});
+
+	test( 'removeFromUploadStarted does nothing when given invalid fieldId', () => {
+		cf2.uploadStarted.push( 'fld456');
+		cf2.uploadStarted.push( 'fld678');
+		removeFromUploadStarted('fld123', cf2 );
+		expect(cf2.uploadStarted.length).toBe(2);
+		expect( cf2.uploadStarted.includes('fld456')).toBe(true);
+		expect( cf2.uploadStarted.includes('fld678')).toBe(true);
+	});
+
+
+
+
+	test( 'removeFromBlocking removes a valid entry when its the only fieldId', () => {
+		cf2.fieldsBlocking.push( 'fld123');
+		removeFromBlocking('fld123', cf2 );
+		expect(cf2.fieldsBlocking.length).toBe(0);
+	});
+
+	test( 'removeFromBlocking removes a valid entry when its one of many fieldId', () => {
+		cf2.fieldsBlocking.push( 'fld456');
+		cf2.fieldsBlocking.push( 'fld123');
+		removeFromBlocking('fld123', cf2 );
+		expect(cf2.fieldsBlocking.length).toBe(1);
+		expect( cf2.fieldsBlocking.includes('fld456')).toBe(true);
+		expect( cf2.fieldsBlocking.includes('fld123')).toBe(false);
+
 	});
 
-	it ( 'finds no fields when searching by invalid field id', () => {
-		expect( getFieldConfigBy(formRenderTestProps.fieldsToControl, 'fieldId', 'fld_1' )).toBe( undefined )
+	test( 'removeFromBlocking does nothing when given invalid fieldId', () => {
+		cf2.fieldsBlocking.push( 'fld456');
+		cf2.fieldsBlocking.push( 'fld678');
+		removeFromBlocking('fld123', cf2 );
+		expect(cf2.fieldsBlocking.length).toBe(2);
+		expect( cf2.fieldsBlocking.includes('fld456')).toBe(true);
+		expect( cf2.fieldsBlocking.includes('fld678')).toBe(true);
+	});
+
+
+	test( 'setBlocking adds to fieldsBlocking', () => {
+		setBlocking('cf123', cf2 );
+		expect(cf2.fieldsBlocking.includes('cf123') ).toBe(true);
+	});
+
+	test( 'setBlocking removes from uploadStarted', () => {
+		cf2.uploadStarted.push( 'cf123');
+		setBlocking('cf123', cf2 );
+		expect(cf2.uploadStarted.includes('cf123') ).toBe(false);
+	});
+
+	test( 'setBlocking removes from pending', () => {
+		cf2.pending.push( 'cf123');
+		setBlocking('cf123', cf2 );
+		expect(cf2.pending.includes('cf123') ).toBe(false);
 	});
 
 });
 
+
+describe( 'createMediaFromFile', () => {
+
+	beforeEach(() => {
+		fetch.resetMocks()
+	});
+
+	const API_FOR_FILES_URL = 'http://localhost:8228/wp-json/cf-api/v3/file';
+	const formId = 'cf1';
+	const nonce = '26bc3db86e';
+	const additionalData = {
+		hashes: ['09164c642f7ae975afe146e7a29d6913'],
```

Comment:
Just a JavaScript note: 

So `const fieldId = 'fl1'; const field = {fieldId:fieldId}` is the same as `const fieldId = 'fl1'; const field = {fieldId}` but shorter.

So you could have used: 
```js
{
		hashes: ['09164c642f7ae975afe146e7a29d6913'],
		verify: '2c6463e902',
		formId,
		fieldId: 'fld_9226671',
		control: 'cf2_file5c1cfbcaa426c',
		_wp_nonce: nonce,
		API_FOR_FILES_URL
	}
```